File: backend/src/main/java/edu/lehigh/cse216/rok224/backend/SimpleRequest.java
package edu.lehigh.cse216.rok224.backend;

/**
 * SimpleRequest provides a format for clients to present title and message 
 * strings to the server.
 * 
 * NB: since this will be created from JSON, all fields must be public, and we
 *     do not need a constructor.
 */
public class SimpleRequest {
    /**
     * The message being provided by the client.
     */
    public String mMessage;
    public String mSessionKey; // maybe refactor to use an HTTP header??
    public String mEmail;
    public jsonFile mFiles;
    public int messageLink;
    public int commentLink;
}

File: android/myapp/lib/simple_request.dart
class SimpleRequest {
  String mMessage;
  String mSessionKey;
  String mEmail;
  //jsonFile mFiles;
  int messageLink;
  int commentLink;

  SimpleRequest(
      {required this.mMessage,
      required this.mSessionKey,
      required this.mEmail,
      //required this.mFiles,
      required this.messageLink,
      required this.commentLink});

  factory SimpleRequest.fromJson(Map<String, dynamic> json) {
    return SimpleRequest(
      mMessage: json['mMessage'] as String,
      mSessionKey: json['mSessionKey'] as String,
      mEmail: json['mEmail'] as String,
      // mFiles: json['mFiles'] as jsonFile,
      messageLink: json['messageLink'] as int,
      commentLink: json['commentLink'] as int,
    );
  }

  Map toMap() {
    var map = new Map();
    map["mMessage"] = mMessage;
    map["mSessionKey"] = mSessionKey;
    map["mEmail"] = mEmail;
    // map["mFiles"] = mFiles;
    map["messageLink"] = messageLink;
    map["commentLink"] = commentLink;

    return map;
  }
}
